job/server/src/test/java/alluxio/job/plan/persist/PersistDefinitionTest.java

```diff
@@ -34,78 +34,66 @@
+    WorkerNetAddress workerNetAddress2 = new WorkerNetAddress().setDataPort(100).setHost("host2");
 
-    long blockId = 1;
-    BlockInfo blockInfo = new BlockInfo().setBlockId(blockId);
-    FileBlockInfo fileBlockInfo = new FileBlockInfo().setBlockInfo(blockInfo);
-    BlockLocation location = new BlockLocation();
-    location.setWorkerAddress(workerNetAddress);
-    blockInfo.setLocations(Lists.newArrayList(location));
+    BlockWorkerInfo blockWorkerInfo1 = new BlockWorkerInfo(workerNetAddress1, 1, 1);
+    BlockWorkerInfo blockWorkerInfo2 = new BlockWorkerInfo(workerNetAddress2, 1, 1);
+
+    WorkerInfo workerInfo1 = new WorkerInfo().setAddress(workerNetAddress1);
+    WorkerInfo workerInfo2 = new WorkerInfo().setAddress(workerNetAddress2);
+
+    FileBlockInfo fileBlockInfo1 = mockFileBlockInfo(1, workerNetAddress2);
+    FileBlockInfo fileBlockInfo2 = mockFileBlockInfo(2, workerNetAddress1);
+    FileBlockInfo fileBlockInfo3 = mockFileBlockInfo(3, workerNetAddress1);
     FileInfo testFileInfo = new FileInfo();
-    testFileInfo.setFileBlockInfos(Lists.newArrayList(fileBlockInfo));
+    testFileInfo.setFileBlockInfos(Lists.newArrayList(fileBlockInfo1, fileBlockInfo2, fileBlockInfo3));
+
+    Mockito.when(mMockFileSystemContext.getCachedWorkers()).thenReturn(
+        Lists.newArrayList(blockWorkerInfo1, blockWorkerInfo2));
     Mockito.when(mMockFileSystem.getStatus(uri)).thenReturn(new URIStatus(testFileInfo));
 
     Set<Pair<WorkerInfo, SerializableVoid>> result =
         new PersistDefinition().selectExecutors(config,
-            Lists.newArrayList(workerInfo), new SelectExecutorsContext(1, mJobServerContext));
+            Lists.newArrayList(workerInfo2, workerInfo1), new SelectExecutorsContext(1, mJobServerContext));
     Assert.assertEquals(1, result.size());
-    Assert.assertEquals(workerInfo, result.iterator().next().getFirst());
+    Assert.assertEquals(workerInfo1, result.iterator().next().getFirst());
   }
 
-  @Test
-  public void selectExecutorsMissingLocationTest() throws Exception {
-    AlluxioURI uri = new AlluxioURI("/test");
-    PersistConfig config = new PersistConfig(uri.getPath(), -1, true, "");
-
-    long blockId = 1;
+  private FileBlockInfo mockFileBlockInfo(long blockId, WorkerNetAddress workerNetAddress) {
     BlockInfo blockInfo = new BlockInfo().setBlockId(blockId);
     FileBlockInfo fileBlockInfo = new FileBlockInfo().setBlockInfo(blockInfo);
-    FileInfo testFileInfo = new FileInfo();
-    testFileInfo.setFileBlockInfos(Lists.newArrayList(fileBlockInfo));
-    Mockito.when(mMockFileSystem.getStatus(uri)).thenReturn(new URIStatus(testFileInfo));
-
-    try {
-      new PersistDefinition().selectExecutors(config,
-          Lists.newArrayList(new WorkerInfo()), new SelectExecutorsContext(1, mJobServerContext));
-    } catch (Exception e) {
-      Assert.assertEquals("Block " + blockId + " does not exist", e.getMessage());
```

Comment:
This test case is useless now. It never run here. I write new unit test as above.